internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -473,14 +484,41 @@ func (mc *mac) generateRouteKeys(stream *livepeer.CreateStreamResp) []string {
 	return keys
 }
 
+func (mc *mac) sign(sharedSecret string) string {
+	s := sha256.New()
+	s.Write([]byte(sharedSecret))
+	signature := hex.EncodeToString(s.Sum(nil))
+	return signature
+}
+
 func (mc *mac) triggerUserNew(w http.ResponseWriter, r *http.Request, lines []string, rawRequest string) bool {
+
 	if len(lines) != 6 {
 		glog.Errorf("Expected 6 lines, got %d, request \n%s", len(lines), rawRequest)
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write([]byte("false"))
 		return false
 	}
 
+	u := &userNewPayload{
+		StreamName:    	lines[0],
+		IpAddress:     	lines[1],
+		UserAgent:     	lines[2],
+		OutputProtocol: lines[3],
+		RequestUrl:    	lines[4],
+		SessionID:     	lines[5],
+	}
+
+	b, err := json.Marshal(u)
+    if err != nil {
+        glog.Errorf("Error marshalling userNewPayload: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("false"))
+        return false
+    }
+
+    payload := string(b)
```

Comment:
**nit:** You can also keep it as `[]byte` and use `bytes.NewReader` or `bytes.NewBuffer` instead (not necessary)